thirdparty/faiss/CMakeLists.txt

```diff
@@ -36,14 +36,11 @@ if (FAISS_ENABLE_CCACHE)
     endif (CCACHE_FOUND)
 endif()
 
+
+
 if(FAISS_ENABLE_GPU)
-  # externalproject_add use semicolon to separate commands, revert to desired format
-  if (CMAKE_CUDA_ARCHITECTURES)
-    string(REPLACE ":" ";" CMAKE_CUDA_ARCHITECTURES "${CMAKE_CUDA_ARCHITECTURES}")
-  endif()
-  set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
   enable_language(CUDA)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_GPU")
+  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -I${FAISS_INCLUDE_DIR} -fPIC")
```

Comment:
CMAKE_CUDA_FLAGS should be passed in from knowhere

Reply:
CMAKE_CUDA_FLAGS is cmake buildin  variables.

Reply:
> CMAKE_CUDA_FLAGS should be passed in from knowhere

CMAKE_CUDA_FLAGS is cmake buildin variables.